src/lib/InstantSearch.ts

```diff
@@ -522,17 +522,18 @@ See ${createDocumentationLink({
     }
   }
 
-  public setUiState: (
-    uiState: UiState | ((uiState: UiState) => UiState)
-  ) => void = uiState => {
+  public setUiState(
+    uiState: UiState | ((uiState: UiState) => UiState),
+    suppressWarning = false
```

Comment:
I'm not sure whether we should just make it non-experimental now and remove the warning completely

Reply:
`setUiState` is 9 month old, if until now we did not get any change to do to this API then I think it is safe to remove this warning and set it as stable.

Reply:
made a subsequent PR, we can discuss there: https://github.com/algolia/instantsearch.js/pull/4597